core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexUpgradeService.java

```diff
@@ -178,12 +178,16 @@ private void checkSupportedVersion(IndexMetaData indexMetaData) {
      * Returns true if this index can be supported by the current version of elasticsearch
      */
     private static boolean isSupportedVersion(IndexMetaData indexMetaData) {
-        if (indexMetaData.creationVersion().onOrAfter(Version.V_0_90_0_Beta1)) {
-            // The index was created with elasticsearch that was using Lucene 4.0
+        if (indexMetaData.creationVersion().onOrAfter(Version.V_2_0_0_beta1)) {
+            // The index was created with elasticsearch that was using Lucene 5.2.1
             return true;
         }
+        if (indexMetaData.getUpgradeVersion().onOrAfter(Version.V_2_0_0_beta1) == false) {
```

Comment:
I'm a bit uncomfortable with change. getUpgradeVersion() relies on `SETTING_VERSION_UPGRADED` which is set when we upgrade lucene segments but *also* when we just upgrade the meta data.  I suggest we remove this for now.